packages/components/src/components/hds/code-block/index.ts

```diff
@@ -31,8 +37,40 @@ import 'prismjs/components/prism-yaml';
 import 'prismjs/components/prism-markup-templating';
 import 'prismjs/components/prism-handlebars';
 
-export default class HdsCodeBlockIndexComponent extends Component {
-  @tracked prismCode = '';
+type Language =
+  | 'bash'
+  | 'go'
+  | 'hcl'
+  | 'json'
+  | 'log'
+  | 'ruby'
+  | 'shell-session'
+  | 'yaml';
+
+export interface HdsCodeBlockIndexSignature {
+  Args: {
+    hasCopyButton?: boolean;
+    hasLineNumbers?: boolean;
+    hasLineWrapping?: boolean;
+    highlightLines?: string;
+    isStandalone?: boolean;
+    language?: Language;
+    maxHeight?: string;
+    value: string;
+  };
+  Blocks: {
+    default: [
+      {
+        Title?: ComponentLike<HdsCodeBlockTitleSignature>;
+        Description?: ComponentLike<HdsCodeBlockDescriptionSignature>;
+      },
+    ];
+  };
+  Element: HTMLDivElement;
+}
+
+export default class HdsCodeBlockIndexComponent extends Component<HdsCodeBlockIndexSignature> {
```

Comment:
suggestion: for brevity and consistency with other components
```suggestion
export default class HdsCodeBlockComponent extends Component<HdsCodeBlockSignature> {
```